app/searchers/application_searcher.rb

```diff
@@ -0,0 +1,113 @@
+class ApplicationSearcher
+  include Enumerable
+
+  class << self
+    attr_accessor :parser_klass, :index_klass, :fields, :keywords, :field_type_mappings
+
+    def index klass
+      @index_klass = klass
+    end
+
+    def keyword key, aliases: nil, as: nil
+      keywords_data << { key => { aliases: aliases, as: as } }
+    end
+
+    def keywords_data
+      @keywords_data ||= []
+    end
+  end
+
+  attr_reader :input
+  attr_reader :errors
+
+  def initialize
+    @errors = Errors.new
+  end
+
+  def query str
+    @input = str
+    self
+  end
+
+  def scope raw_es_query
+    @scope = raw_es_query
+    self
+  end
+
+  def results
+    raw_results = index_klass.query build_elasticsearch_query.to_elasticsearch
+    raw_results.query @scope
+    fetch raw_results
+  end
+
+  def each
+    results.each
+  end
+
+  protected
+
+  def index_klass
+    self.class.index_klass
+  end
+
+  def keywords_data
+    self.class.keywords_data
+  end
+
+  def keywords
+    self.class.keywords ||= keywords_data.flat_map do |keyword_data|
+      keyword = keyword_data.keys.first
+      data = keyword_data[ keyword ]
+
+      next as if data[:as].present?
```

Comment:
is `as` relevant in this scope?

Reply:
Left over, but means I could do like
```ruby
keyword :superlong_field, as: :short
```
But I haven't ended up needing it yet, so I did consider removing this little bit